src/ga_session.cpp

```diff
@@ -2795,6 +2797,16 @@ namespace sdk {
             GDK_RUNTIME_ASSERT(server_address == user_address);
         }
 
+        if (addr_type == address_type::csv) {
+            locker_t locker(m_mutex);
```

Comment:
nit: can this lock be moved below the call to get_csv_blocks_from_csv_redeem_script?